### DISCUSSION TITLE
Palm Pilot Zire 21 connection to Debian 12

### DISCUSSION THREAD

**Discussion by idubinsky9** [desrod/pilot-link#20] [Q&A]
Folks, does somebody have an experience with connection of Palm Pilot Zire 21 to Debian 12. Idea is later to use it with AQEMU Windows 10 VM (Palm desktop installation on it.)

Thank you in advance!

**Comment by desrod** [comment]
@idubinsky9 A Palm device doesn't maintain a persistent connection when connected via the USB cable, until you initiate a HotSync operation. Then it will create a tty device, normally `/dev/ttyUSB1`, which you can connect to with various apps. 

If you're using an emulator,  you need to make sure you pass that device into the VM as a serial port, so the VM "owns" that port each time it connects (and not your Linux host). Then from inside the Windows 10 VM, you should be able to see, detect and use the Palm Zire 21 from there. 

Good luck!

**Reply by colinstu12** [reply]
One trouble I've run into, is that this TTY device only displays momentarily, and if I'm passing it to a VM on Proxmox, I can't start up a VM if said device doesn't actually appear / exist yet. I know this is out of your control, but just reporting my findings / may want to avoid proxmox for this scenario. (Unless anyone knows a way to handle this..). Proxmox makes use of KVM/QEMU, so perhaps this issue of not being able to start up a VM if you configure it to share that TTY device (but it's not currently present) extends beyond it too.

**Reply by desrod** [reply]
> One trouble I've run into, is that this TTY device only displays momentarily, and if I'm passing it to a VM on Proxmox, I can't start up a VM if said device doesn't actually appear

What does your udev rule on the Proxmox VM side look like? And are you attaching a real serial port to the VM in Proxmox's settings?

**Reply by colinstu12** [reply]
> What does your udev rule on the Proxmox VM side look like? And are you attaching a real serial port to the VM in Proxmox's settings?

I'm using my Visor's cradle/dock, which is physically USB (but is presented as Serial to the computer, but only for a split second when the sync button is being pressed / HotSync process is happening).

**Reply by desrod** [reply]
> I'm using my Visor's cradle/dock, which is physically USB (but is presented as Serial to the computer, but only for a split second when the sync button is being pressed / HotSync process is happening).

Just so I can try to reliably reproduce this: You have your USB cradle physically connected to your Proxmox host, and then you pass that USB serial port into the VM as an attached serial port? 

Inside that VM, what does your Palm udev rule look like for your Zire 21? It will be under `/etc/udev/rules.d/` inside the VM.
